subsys/net/l2/ethernet/dsa/dsa_user.c

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(net_dsa_user, CONFIG_NET_DSA_LOG_LEVEL);
+
+#include <zephyr/net/ethernet.h>
+#include <zephyr/net/dsa_core.h>
+
+struct net_if *dsa_user_get_iface(struct net_if *iface, int port_idx)
+{
+	struct ethernet_context *eth_ctx = net_if_l2_data(iface);
+	struct dsa_switch_context *dsa_switch_ctx;
+
+	if (!eth_ctx || !eth_ctx->dsa_switch_ctx) {
```

Comment:
We try to avoid this style of checks and have been converting these to explicit NULL checks in other parts of the code. So better to do here
```
if (eth_ctx == NULL || eth_ctx->dsa_switch_ctx == NULL) {
```
Please check other such invocations in this PR and adjust the code accordingly.

Reply:
Thanks, will fix all.